code/config/lib/Loot/Config/Record.hs

```diff
@@ -133,7 +202,17 @@ type family SubRecsConstrained c k is :: Constraint where
         , SubRecsConstrained c k us
         , SubRecsConstrained c k is
         )
-
+    SubRecsConstrained c k ((l ::+ us) ': is) =
+        ( c (ConfigRec k (SumSelection l : ToBranches us))
+        , SubRecsConstrained c k (SumSelection l : ToBranches us)
+        , SubRecsConstrained c k is
+        )
+    SubRecsConstrained c k ((_ ::- us) ': is) =
+        ( c (ConfigRec k us)
+        , c (Maybe (ConfigRec k us))
```

Comment:
You only need `c (ConfigRec k us)` if `k ~ 'Partial` and `c (Maybe (ConfigRec k us))` if `k ~ 'Final`. This should be split in two pattern matches, matching also by `k`.

Reply:
It is right that I only need `c (ConfigRec k us)` if `k ~ 'Partial` (so I did split it in two pattern matches), but I don't think that `c (Maybe (ConfigRec k us))` is enough when `k ~ 'Final`.

Take the `Show` instance for `Maybe`: it is `Show a => Show (Maybe a)`, it would be enough if we only use Show on a `Maybe (ConfigRec k us)`, but the type system cannot guarantee that `ConfigRec k us` is part of `Show` as well.
We actually use this (in the `Show (Item k i)` instance definition), so we actually need to impose `Show` on `c (ConfigRec k us)` as well.

Also, this case seems to be very common, so I left it for now, tell me what you think.

Reply:
@pasqu4le Actually, I just tried to remove constraint `c (Maybe (ConfigRes k us))` altogether, leaving only this:
```
    SubRecsConstrained c k ((_ ::- us) ': is) =
        ( c (ConfigRec k us)
        , SubRecsConstrained c k us
        , SubRecsConstrained c k is
        )
```
and it works just fine, analogously with how `c v` without `c (Maybe v)` constraint works just fine for any kind of configs in `ValuesConstrained`.

Reply:
You are right, the constraint on the`Maybe` never get used and it's not necessary at all. Removed.